### DISCUSSION TITLE
RTL (Right-to-Left) Support

### DISCUSSION THREAD

**Discussion by shlomtubul5** [toeverything/AFFiNE#8141] [Ideas]
I would like to request support for RTL (Right-to-Left) languages in this application. RTL support is essential for users who speak and write in languages such as Hebrew, Arabic, Persian, and others. Without proper RTL support, the user experience can become difficult due to misaligned text, incorrect interface layout, and overall poor usability.

By adding RTL support, this app will become more accessible to a wider audience, enhancing the experience for users who rely on these languages.

Thank you for considering this feature request. I believe it will greatly improve the usability for many people around the world.

**Comment by zoh-li** [comment]
This project is only lacking RTL support.

**Comment by davidulman** [comment]
+1

**Comment by zetashift** [comment]
If you go into `Settings > Experimental features` you can toggle experimental RTL support:

![image](https://github.com/user-attachments/assets/ce13c83c-5730-4d3d-9e83-0428cc095fb0)


And if something is broken an issue can be reported(hopefully with a reproduction case).

**Reply by danial-kazmi** [reply]
Using Desktop application, I can't see this option,
would you please mention where you see this?
<img width="642" height="637" alt="{B06E215D-FFC2-4C9F-9E8F-46102E30931C}" src="https://github.com/user-attachments/assets/efb33f6e-2275-439a-8bbd-a4b9f333edcc" />

**Reply by alirexa** [reply]
I'm self hosting via Docker and I don't see the RTL toggle in the experimental features too.

**Comment by drorata** [comment]
I guess this is the right place to upvote: https://github.com/toeverything/blocksuite/issues/7894

**Comment by rellikivo-ai** [comment]
We need RTL support for the fantastic app AFFINE ❤️
